src/clover/analysis.py

```diff
@@ -230,6 +235,7 @@ def get_key_results(
 def plot_outputs(  # pylint: disable=too-many-locals, too-many-statements
     grid_input_profile: pd.DataFrame,
     grid_profile: Optional[pd.DataFrame],
+    grid_profiles: Optional[Dict[str, pd.DataFrame]],  # to check that
```

Comment:
@paulharfouche , you're getting an error because you've kept both `grid_profile` and `grid_profiles` here, which means that the function is expecting one more argument than it's being given, hence the error

Reply:
Hi @BenWinchester, but the second I take grid_profile or grid_profiles out... I get this error which makes sense:
![image](https://user-images.githubusercontent.com/106335872/176174145-d8c8c9df-c4cd-4c7a-9c33-6d859acb7a3e.png)

Reply:
@BenWinchester In case I take the grid_profile out I get this:
![image](https://user-images.githubusercontent.com/106335872/176175186-78f9d5f3-0c3b-4e0a-8668-33371ca73282.png)

Reply:
Hi @paulharfouche , by taking it out, you're doing the right thing :+1: However, you'll need to update the function itself:

### With the argument
The function is happy to use the term `grid_profile` in the `plot_outputs` function, but..
* We don't want this, as we now have multiple grid profiles to deal with,
* and the function won't be executed as the arguments don't match.

### Without the argument
The function will be callable, which is what's happening here in the traceback: each line that you see in the traceback (that's the output you're seeing in the terminal) is a place where Python managed to get to as it executed the code. So, here, it got to line 1062 in the `main` function of the `__main__` module, and line 305 in the `analysis` module in the `plot_outputs` function: so it's getting further than before, into the function.
The line of the code that's causing the error is also shown:
```python
total=(9 + (1 if grid_profile is not None else 0)),
```
This line is part of the progress bar, determining the number of plots that need to be generated so that the `tqdm` progress bar can display correctly. Now that we have the `grid_profiles` vairable, the `grid_profile` variable no longer exists, and so this line can't exist. You'd get the same error if you typed
```python
>>> grid_profile
```
straight into a Python shell.

So, there are several ways this could be fixed:
1. We just delete the reference: this isn't a particularly important part of the code, so we could just have
   ```python
   total = 9
   ```
2. We update this to include the fact that we have multiple grid profiles: this is the best solution
   ```python
   total = (9 + (len(grid_profiles) if grid_profiles is not None else 0)),
   ```
   i.e., we want to use the number of grid profiles in our plots. This is what's actually happening, by including the `for grid_profile in grid_profiles` loop for generating the heat map plots, CLOVER will be plotting 9 plots, plus one for every grid profile that we want it to plot :smile:

Reply:
Oh okay! but the for grid_profile in grid_profiles is being read later right?
![image](https://user-images.githubusercontent.com/106335872/176178566-9bc2b540-f44e-422d-8c09-e7878cc66804.png)

Previously I had it before `with tqdm` and you told me to move it to where it is now...
The problem is 

    ```
    for grid_profile in grid_profiles:
            # Plot the grid availablity profile.
            if grid_profile is not None:
                reshaped_data = np.reshape(
                    grid_profile.iloc[0:HOURS_PER_YEAR].values, (365, 24)
                )
                heatmap = sns.heatmap(
                    reshaped_data, vmin=0, vmax=1, cmap="Greys_r", cbar=False
                )
```
Is actually reading grid_profile as str and is telling me this error.... 
![image](https://user-images.githubusercontent.com/106335872/176178887-6a57b368-d537-4b22-a006-f06c58175b6c.png)